packages/gateway/src/manager.ts

```diff
@@ -99,6 +119,12 @@ export function createGatewayManager(options: CreateGatewayManagerOptions): Gate
         if (typeof info.lastShardId === 'number') gateway.lastShardId = info.lastShardId
         gateway.logger.warn(`[Resharding] Starting the reshard process. New Total Shards. ${gateway.totalShards}`)
 
+        if (oldShardNumber === gateway.totalShards) {
+          gateway.logger.warn('[Resharding] Aborting the reshard process. The total number of shards is the same')
+
+          return
+        }
+
```

Comment:
```suggestion

```

Reply:
This is not needed and shouldn't be included so you could use resharding as a way to change your gateway versions in gateway proxy (this was supposed to be one of the usecases even tho it might not be too easy to do)

Reply:
> as a way to change your gateway versions in gateway proxy

might be me: but that does like a dumb use-case to me lol, aside from that, fine ig?